lib/ansible/module_utils/network/ios/config/vlans/vlans.py

```diff
@@ -146,16 +146,22 @@ def _state_overridden(self, want, have, state):
         """
         commands = []
 
+        want_local = want
         for each in have:
-            for every in want:
+            count = 0
+            for every in want_local:
                 if each['vlan_id'] == every['vlan_id']:
                     break
+                count += 1
             else:
                 # We didn't find a matching desired state, which means we can
                 # pretend we recieved an empty desired state.
                 commands.extend(self._clear_config(every, each, state))
                 continue
             commands.extend(self._set_config(every, each))
+            del want_local[count]
```

Comment:
does want_local has a count key already? what is it used for?

Reply:
@danielmellado I have introduced `count` variable coz with the current approach only pre-existing VLAN will be matched and then once matched respective VLAN will be configured and since the count is keeping the track of dict in list, once configured I'll delete the respective entry from the list, so that only new VLAN are left in want list and I can configure those directly.

Reply:
I'll write a comment for more code readability

Reply:
Mind adding a comment to keep track of this? I think the future us will be thankful ;)

Reply:
Thanks @danielmellado and have updated the code comment. :)